devtools/platform-descriptor-json/src/main/resources/templates/dockerfile-fast-jar.ftl

```diff
@@ -45,10 +45,7 @@ RUN microdnf install curl ca-certificates ${JAVA_PACKAGE} \
 # Configure the JAVA_OPTIONS, you can add -XshowSettings:vm to also display the heap size.
 ENV JAVA_OPTIONS="-Dquarkus.http.host=0.0.0.0 -Djava.util.logging.manager=org.jboss.logmanager.LogManager"
 
-COPY --chown=1001 ${build_dir}/quarkus-app/lib/ /deployments/lib/
-COPY --chown=1001 ${build_dir}/quarkus-app/*.jar /deployments/
-COPY --chown=1001 ${build_dir}/quarkus-app/app/ /deployments/app/
-COPY --chown=1001 ${build_dir}/quarkus-app/quarkus/ /deployments/quarkus/
+COPY --chown=1001 ${build_dir}/quarkus-app /deployments/
```

Comment:
Is it equivalent? I would expect this to copy the `quarkus-app` directory inside `/deployments/` whereas we used to copy the content of this directory, meaning the main jar was at the root of `/deployments/`.

At least if Docker's `COPY` has the same semantic as `cp`.

Reply:
I tested it locally by adding a `RUN ls /deployments` line after that and the log shows that the contents of the directory have been copied:

```
Step 13/17 : COPY --from=builder --chown=1001 /usr/src/app/bff/build/quarkus-app /deployments/
 ---> Using cache
 ---> 61e79dc8db06
Step 14/17 : RUN ls /deployments
 ---> Running in 5124e52a774d
app
lib
quarkus
quarkus-run.jar
run-java.sh
```

Reply:
OK. That's a bit weird IMO but not familiar with Docker so I trust you :).

That being said, I'm wondering if we could change that to:
```
COPY --chown=1001 ${build_dir}/quarkus-app/* /deployments/
```
to be on the safe side?

Reply:
Unfortunately this seems to unpack the directories in the quarkus-app directory. The `ls` now shows

```
app-1.0.0-SNAPSHOT.jar
boot
generated-bytecode.jar
main
quarkus-application.dat
quarkus-run.jar
run-java.sh
transformed-bytecode.jar
```

Reply:
@gsmet: In the [docker copy docs](https://docs.docker.com/engine/reference/builder/#copy), I just found a note saying the following about the src path:

```
The directory itself is not copied, just its contents.
```

Furthermore, the pattern matching seems to the one Golang uses. Therefore I think it doesn't apply the standard linux semantics.